### PR TITLE
 ncm-metaconfig: add irods config files

### PR DESCRIPTION
**Pull request by kwaegema**
Requires quattor/CAF#265
Requires quattor/CCM#186

### CODE REVIEW

**Comment by kwaegema** [comment]
depends on https://github.com/quattor/CAF/pull/265

**Comment by kwaegema** [comment]
also on https://github.com/quattor/CCM/pull/186

**Comment by jrha** [comment]
Please add requirements to the PR description (not as comments) on separate lines otherwise the backlog parser will not find them.

**Comment by jrha** [comment]
@hpcugentbot retest this please

**Comment by jrha** [comment]
Moved milestone to after @kwaegema is back from holiday

**Comment by kwaegema** [comment]
This requires the CCM changes to work

**Comment by jrha** [comment]
@hpcugentbot test this please

File: ncm-metaconfig/src/main/metaconfig/irods/pan/schema.pan [hunk 1]
```diff
@@ -0,0 +1,73 @@
+declaration template metaconfig/irods/schema;
+
+include 'pan/types';
+
+@{ type for configuring the irods_environment.json config file for clients @}
+type irods_environment_client_config = {
+    'irods_host' :  type_hostname
```
Comment (Pansanel) [inline ncm-metaconfig/src/main/metaconfig/irods/pan/schema.pan]: Minor syntax suggestions:
* Remove space before ':'
* Replace ' by " (like most of the other lines)

File: ncm-metaconfig/src/main/metaconfig/irods/pan/schema.pan [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+};
+@{ type for configuring the irods_environment.json config file for service accounts @}
+type irods_environment_server_config = {
+    include irods_environment_client_config
+    "irods_cwd": string = "/UGent/home/rods"
+    "irods_default_hash_scheme": choice('SHA256', 'MD5') = "SHA256"
+    "irods_default_number_of_transfer_threads": long =  4
+    "irods_default_resource": string = "demoResc"
+    "irods_maximum_size_for_single_buffer_in_megabytes": long = 32
+    "irods_home": string = "/UGent/home/rods"
```
Comment (Pansanel) [inline ncm-metaconfig/src/main/metaconfig/irods/pan/schema.pan]: May be better not to set a default here (not all sites have a 'UGent' home ;-) )

File: ncm-metaconfig/src/main/metaconfig/irods/tests/profiles/access_control.pan [hunk 3]
```diff
@@ -0,0 +1,7 @@
+object template access_control;
+
+include 'metaconfig/irods/access_control';
+
+
```
Comment (Pansanel) [inline ncm-metaconfig/src/main/metaconfig/irods/tests/profiles/access_control.pan]: a single empty line should be enough

File: ncm-metaconfig/src/main/metaconfig/irods/tests/profiles/config.pan [hunk 4]
```diff
@@ -0,0 +1,17 @@
+object template config;
+
+include 'metaconfig/irods/config';
+
+bind "/software/components/metaconfig/services/{/etc/nagios/irods/irods_environment.conf}/contents" = irods_environment_client_config;
+
+prefix "/software/components/metaconfig/services/{/etc/nagios/irods/irods_environment.conf}";
+"module" = "jsonpretty";
+
+
```
Comment (Pansanel) [inline ncm-metaconfig/src/main/metaconfig/irods/tests/profiles/config.pan]: a single empty line should be enough

File: ncm-metaconfig/src/main/metaconfig/irods/tests/profiles/config.pan [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+prefix "/software/components/metaconfig/services/{/etc/nagios/irods/irods_environment.conf}/contents";
+
+"irods_host" = "iicat01.ugent.be";
```
Comment (Pansanel) [inline ncm-metaconfig/src/main/metaconfig/irods/tests/profiles/config.pan]: We should avoid to set such hostname. Something more anonymous would be better.

Comment (jrha) [reply ncm-metaconfig/src/main/metaconfig/irods/tests/profiles/config.pan]: icat01.example.org?

**Comment by kwaegema** [comment]
@Pansanel cleanup done :)

**Pull request merged**